### PR TITLE
feat: Added separate project config endpoint

### PR DESCRIPTION
**Pull request by mitsuhiko**

### CODE REVIEW

**Comment by mitsuhiko** [comment]
This should not be merged yet until we are happy with the design given the migration.

File: src/sentry/api/endpoints/relay_register.py [hunk 1]
```diff
@@ -34,20 +34,21 @@ class RelayRegisterResponseSerializer(RelayIdSerializer):
     token = serializers.CharField(required=True)
 
 
+def is_internal_relay(request, public_key):
+    """
+    Checks if the relay is allowed to register, otherwise raises an exception
```
Comment (jan-auer) [inline src/sentry/api/endpoints/relay_register.py]: _nit_: This comment is misleading, the function does not raise.

File: src/sentry/relay/config.py [hunk 2]
```diff
@@ -1,50 +1,59 @@
...
+def get_organization_options(org):
+    return {
+        'trustedRelays': org.get_option('sentry:trusted-relays', []),
+    }
+
+
+def relay_has_org_access(relay, org):
+    # Internal relays always have access
+    if relay.is_internal:
+        return True
```
Comment (jan-auer) [inline src/sentry/relay/config.py]: To avoid the migration you could just check for the `WHITELIST_PK` here.

**Review by jan-auer — APPROVED** [review]

**Comment by untitaker** [comment]
I don't understand the purpose of having all three of is_internal + PK whitelist + IP whitelist. Seems like the latter two should suffice.

**Comment by mitsuhiko** [comment]
The `trustedRelays` moved into the config. Will figure out how to bump the hash based on org config changes.

@untitaker not sure I understand to which code the comment longs to

**Comment by mitsuhiko** [comment]
WRT to the `is_internal` flag. I'm not fully convinced into either storing or deriving it. However if it's not stored it's kinda hard to query it.

**Comment by mitsuhiko** [comment]
@jan-auer can you have another look and i will get this merged

**Comment by jan-auer** [comment]
Let me run a manual test briefly

**Review by jan-auer — APPROVED** [review]
Verified, let's just make sure to roll out relays soon after. Let's merge this.

**Pull request merged**